crates/storage/provider/src/trie/mod.rs

```diff
@@ -400,6 +494,48 @@ impl DBTrieLoader {
 
         Ok(hashed_changes)
     }
+
+    /// Returns a Merkle proof of the given account, plus its storage root hash.
+    pub fn generate_acount_proof<'tx, 'itx>(
+        &self,
+        tx: &'tx impl DbTx<'itx>,
+        root: H256,
+        address: H256,
+    ) -> Result<(Vec<Vec<u8>>, H256), TrieError> {
+        let db = Arc::new(HashDatabase::from_root(tx, root)?);
+        let hasher = Arc::new(HasherKeccak::new());
+
+        let trie = PatriciaTrie::from(Arc::clone(&db), Arc::clone(&hasher), root.as_bytes())?;
+        let proof = trie.get_proof(address.as_bytes())?;
+
+        let Some(account) = trie.get(address.as_slice())? else { return Ok((proof, KECCAK_EMPTY)) };
+
+        let storage_root = EthAccount::decode(&mut account.as_slice())?.storage_root;
+
+        Ok((proof, storage_root))
+    }
+
+    /// Returns a Merkle proof of the given storage keys, starting at the given root hash.
+    pub fn generate_storage_proofs<'tx, 'itx>(
+        &self,
+        tx: &'tx impl DbTx<'itx>,
+        storage_root: H256,
+        address: H256,
+        keys: Vec<H256>,
+    ) -> Result<Vec<Vec<Vec<u8>>>, TrieError> {
```

Comment:
Vec<Vec<Vec<_>>> looks weird, can we introduce a new type for this?

Reply:
Added a type alias for `Vec<Vec<u8>>` named `MerkleProof`